web/src/message_edit.ts

```diff
@@ -1186,14 +1258,15 @@ export function delete_message(msg_id) {
     });
 }
 
-export function delete_topic(stream_id, topic_name, failures = 0) {
-    channel.post({
+export function delete_topic(stream_id: number, topic_name: string, failures = 0): void {
+    void channel.post({
         url: "/json/streams/" + stream_id + "/delete_topic",
         data: {
             topic_name,
         },
         success(data) {
-            if (data.complete === false) {
+            const {complete} = z.object({complete: z.boolean().nullable()}).parse(data);
```

Comment:
This doesn’t seem to be nullable.

https://github.com/zulip/zulip/blob/52a9846cdf4abfbe937a94559690d508e95f4065/zerver/views/streams.py#L900-L936